pkg/cloudproviders/aws/aws.go

```diff
@@ -74,28 +76,45 @@ func (p *Provider) RunMetrics(ctx context.Context, statser stats.Statser) {
 // map is returned even in case of errors because it may contain partial data.
 func (p *Provider) Instance(ctx context.Context, IP ...gostatsd.Source) (map[gostatsd.Source]*gostatsd.Instance, error) {
 	instances := make(map[gostatsd.Source]*gostatsd.Instance, len(IP))
-	values := make([]*string, len(IP))
+	values := make([]string, len(IP))
 	for i, ip := range IP {
 		instances[ip] = nil // initialize map. Used for lookups to see if info for IP was requested
-		values[i] = aws.String(string(ip))
+		values[i] = string(ip)
 	}
-	input := &ec2.DescribeInstancesInput{
-		Filters: []*ec2.Filter{
-			{
-				Name:   aws.String("private-ip-address"),
-				Values: values,
-			},
+	privateIPFilter := "private-ip-address"
+	inputFilters := []ec2Types.Filter{
+		{
+			Name:   &privateIPFilter,
+			Values: values,
 		},
 	}
 
 	atomic.AddUint64(&p.describeInstanceCount, 1)
 	atomic.AddUint64(&p.describeInstanceInstances, uint64(len(IP)))
 	instancesFound := uint64(0)
 	pages := uint64(0)
+	var err error
+	var nextToken string
 
 	p.logger.WithField("ips", IP).Debug("Looking up instances")
-	err := p.Ec2.DescribeInstancesPagesWithContext(ctx, input, func(page *ec2.DescribeInstancesOutput, lastPage bool) bool {
+	hasPagesRemaining := true
+	for hasPagesRemaining {
```

Comment:
I wonder can we use `NewDescribeInstancesPaginator` for this?

Reply:
We probably can, this is an instance of the migrated functions equivalent not being obvious for the writer

Reply:
Nice saw the changes